### ISSUE TITLE
browser-generated (PAGEVIEW) telemetry doesn't correlate with server-generated telemetry

### ISSUE THREAD

**Issue by jhsheets** [microsoft/ApplicationInsights-Java#821]
I'm unable to get browser-generated telemetry to correlate with the server-generated telemetry that occurs as a results of a pageview. All the telemetry is displayed in AI, but they don't correlate.

I've created an example application that showcases the issue:
https://github.com/jhsheets/java-app-insight-not-correlating-pageview

**Comment by dhaval24** [comment]
Thanks for creating the issue @jhsheets . We will triage this in our coming releases.

**Comment by dhaval24** [comment]
@jhsheets a quick questions: 

1. which version of Application Insights Java script sdk do you use?  For correlation to happen the incoming request should have a request-Id header with correlation-id present. 

I have not got a chance to look at your entire application but on high level it seems : 

a. you are using AzureStorage SDK, and manually tracking the storage dependency with `TelemetryClient.trackDependency()`.

The flow is like this: Browser (enabled with AI JS sdk) sends request -> Springboot controller gets request -> calls azure storage (Tracked manually) -> returns response. 

Some side line code comments:

https://github.com/jhsheets/java-app-insight-not-correlating-pageview/blob/5baecba5e839b70e500f3a3d5c1c2081eb20da98/src/main/java/com/example/filestore/AzureSoftwareFileStorage.java#L33

Here you should use @Autowire to get the correctly configured TelemetryClient, unless you need some special configuration that you have created yourself.

https://github.com/jhsheets/java-app-insight-not-correlating-pageview/blob/5baecba5e839b70e500f3a3d5c1c2081eb20da98/src/main/java/com/example/metrics/MyTelemetryProcessor.java#L19

You should wire this TelemetryProcessor as a Bean to be used correctly.

https://github.com/jhsheets/java-app-insight-not-correlating-pageview/blob/master/src/main/java/com/example/metrics/AzureAppInsightsContextListener.java

I am not able to fully understand the purpose of this context listener. 

cc: @markwolff @jpiyali for AI-JS SDK.

**Comment by jhsheets** [comment]
I followed the instructions from https://github.com/Microsoft/ApplicationInsights-JS/blob/master/API-reference.md and copied the non-minified code from https://az416426.vo.msecnd.net/scripts/a/ai.js

I wrote this example app weeks ago, but It looks like they're still identical.


> Here you should use @autowire to get the correctly configured TelemetryClient, unless you need some special configuration that you have created yourself.

It is autowired. There's a setter below the constructor for it. It's been a while since I wrote this example app, but I think I didn't inject into the constructor because it made it easier for me to write integration tests where I didn't want to track telemetry.

> Here you should use @autowire to get the correctly configured TelemetryClient, unless you need some special configuration that you have created yourself.

It is a `@Bean`. The AzureAppInsightsContextListener.java file is a `@Configuration `class that makes it available. That class has some old code (before the spring-boot-starter lib existed) that I commented out.
